src/preferArrayLiteralRule.ts

```diff
@@ -6,16 +6,44 @@ import { AstUtils } from './utils/AstUtils';
 import { ExtendedMetadata } from './utils/ExtendedMetadata';
 import { isObject } from './utils/TypeGuard';
 
+// undefined for case when function/constructor is called directly without namespace
+const RESTRICTED_NAMESPACES = [undefined, 'window', 'global', 'globalThis'];
```

Comment:
> `self = this`

IMO `"self"` should be included here. There's a separate lint rule to exclude the `var self = this;` pattern, which is a holdover from old now-unnecessary (and hopefully now uncommon) styles.

Reply:
OK. Wan't aware about rule.